src/transaction/transaction_manager.cpp

```diff
@@ -45,6 +45,7 @@ void TransactionManager::LogCommit(TransactionContext *const txn, const timestam
 
 timestamp_t TransactionManager::ReadOnlyCommitCriticalSection(TransactionContext *const txn, const callback_fn callback,
                                                               void *const callback_arg) {
+  TERRIER_ASSERT(txn->commit_actions_.empty(), "Only updating transactions can have deferred actions");
```

Comment:
This might not be true in the future. (For example, callback from logging can actually be replaced with this deferred action as well, eliminating work on the logging thread) We can either leave it here with a comment that this may change in the future or remove it. Up to you.

Reply:
If we allow read-only transactions to trigger actions, then we need to reason about how much of this function needs to be atomic.  The current code assumes that read-only transactions have no impact on the actual database and therefore do not need the commit latch.

My inclination is to comment/document the current assumption, unless someone else weighs in.

Reply:
Yeah that sounds like you should write a TODO

Reply:
After talking with @mbutrovich, it made more sense to move the commit actions to after the critical section in commit.  Making this change negated the need for the constraint.